### PR TITLE
nf: Undeprecate AnkiDroidApp.getSharedPrefs

### PR DESCRIPTION
**Pull request by BrayanDSO**
The androidx version of the method is equal to the android one, so this is actually a non-functional change

## How Has This Been Tested?

Open the app and change some prefs

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by lukstbit — APPROVED** [review]
LGTM

**Comment by BrayanDSO** [comment]
A refactor idea here would be to create a `Context.getSharedPrefs()` method, then in-line and replace all usages of `AnkiDroidApp.getSharedPrefs()`

It should take a couple of minutes in the IDE, but it would need a quick merge to avoid conflicts, so I'll leave the idea pending

**Comment by lukstbit** [comment]
> A refactor idea here would be to create a Context.getSharedPrefs() method, then in-line and replace all usages of AnkiDroidApp.getSharedPrefs()

I like the idea. If you want to do it, go for it and I'll review and merge directly.
Just keep the original contexts so we don't have any surprises and maybe we could also drop the `get` part and simply use `Context.sharedPrefs()`(?!).

File: AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt [hunk 1]
```diff
@@ -433,7 +435,13 @@ fun DeckPicker.createSyncListener(isFetchingMedia: Boolean) = object : Connectio
                     resources.getString(R.string.sync_title),
                     """
                                 ${resources.getString(R.string.sync_title)}
-                                ${resources.getString(R.string.sync_up_down_size, mCountUp, mCountDown)}
+                                ${
+                    resources.getString(
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt:439]: There's no need for this change.

File: AnkiDroid/src/test/java/com/ichi2/anki/InitialActivityTest.kt [hunk 2]
```diff
@@ -45,7 +46,7 @@ class InitialActivityTest : RobolectricTest() {
 
     @Before
     fun before() {
-        mSharedPreferences = AnkiDroidApp.getSharedPrefs(ApplicationProvider.getApplicationContext())
+        mSharedPreferences = ApplicationProvider.getApplicationContext().sharedPrefs()
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/InitialActivityTest.kt:49]: This `ApplicationProvider.getApplicationContext()` call and the others in this file require a `Context` type parameter.

**Review by lukstbit — COMMENTED** [review]
Went through and everything seems fine.
Do you enabled format on save? There are some formatting changes bundled here as well.

**Comment by BrayanDSO** [comment]
> There are some formatting changes bundled here as well.

Unintentional. The "inline" tool did it. Idk if I have to disable some setting somewhere to avoid it

**Review by lukstbit — APPROVED** [review]
Ok! As agreed I'm merging this directly.

**Pull request merged**